src/envoy/extensions/wasm/wasm.cc

```diff
@@ -101,15 +102,23 @@ class IstioWasmExtension : public EnvoyWasm {
 };
 
 WasmHandleExtensionFactory IstioWasmExtension::wasmFactory() {
-  return [](const VmConfig vm_config, const Stats::ScopeSharedPtr& scope,
+  return [](const VmConfig vm_config,
+            const CapabilityRestrictionConfig capability_restriction_config,
+            const Stats::ScopeSharedPtr& scope,
             Upstream::ClusterManager& cluster_manager,
             Event::Dispatcher& dispatcher,
             Server::ServerLifecycleNotifier& lifecycle_notifier,
             absl::string_view vm_key) -> WasmHandleBaseSharedPtr {
-    auto wasm =
-        std::make_shared<IstioWasm>(vm_config.runtime(), vm_config.vm_id(),
-                                    anyToBytes(vm_config.configuration()),
-                                    vm_key, scope, cluster_manager, dispatcher);
+    // TODO(rapilado): make this transformation in Proxy-Wasm C++ Host.
```

Comment:
@ryanapilado it looks that this leaks quite a bit and each extension needs to convert `CapabilityRestrictionConfig` to `allowed_capabilities`, which seems a bit error prone. We should do it earlier in the process (either in Proxy-Wasm C++ host or Envoy) to avoid this repetition.

cc @mathetake to catch similiar issues in the future.